### PR TITLE
[UX/UI] Some fixes in search autocomplete

### PR DESCRIPTION
**Pull request by arielj**
This PR includes a fix for https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/issues/2575 and some other fixes I noticed while fixing that:

- if we had the same game in 2 stores, there was no way to select which one to open and Epic had priority (this was not an issue when clicking only changed the results), I added a `(GOG)` or `(Epic)` or `(sideload)` suffix
- we were ordering and then filtering, which is slower than first filtering and then ordering since then we need to order less elements
- we were converting the list of games to a list of titles to render it and then on click we were fetching the game again, now it just uses the list of games to render it and passes the game as an argument on click to avoid that extra work
- since now we only have unique games, no need to convert everything into a set and then back to an array, so now it's all a single array
- it now ignores DLCs from the search result, which created issues (before it was producing the error in the issue linked, after the element changed to open the game page it was opening the DLC page which doesn't really work)

image

---

Use the following Checklist if you have changed something on the Backend or Frontend:

- [ ] Tested the feature and it's working on a current and clean install.
- [ ] Tested the main App features and they are still working on a current and clean install. (Login, Install, Play, Uninstall, Move games, etc.)
- [ ] Created / Updated Tests (If necessary)
- [ ] Created / Updated documentation (If necessary)

### CODE REVIEW

**Comment by flavioislima** [comment]
Can you add a small fix, probably one line, to the shortcuts creator? I think Dlcs after being installed are cresting shortcuts and being added to steam and desktop if these options are on on the settings.

So just need to check if it's a dlc before adding the shortcuts. 👍

**Comment by arielj** [comment]
> Can you add a small fix, probably one line, to the shortcuts creator? I think Dlcs after being installed are cresting shortcuts and being added to steam and desktop if these options are on on the settings.
> 
> So just need to check if it's a dlc before adding the shortcuts. +1

I'll do that in a different PR, there's also an error related to shortcuts for sideloaded apps and that change will make more sense there https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/issues/2756

File: src/frontend/components/UI/SearchBar/index.tsx [hunk 1]
```diff
@@ -30,19 +35,19 @@ export default React.memo(function SearchBar() {
   const list = useMemo(() => {
     // Set can't handle spread of undefined. Leading to
```
Comment (flavioislima) [inline src/frontend/components/UI/SearchBar/index.tsx:36]: I think you can remove these comments now.

**Review by flavioislima — APPROVED** [review]
LGTM

File: src/frontend/components/UI/SearchBar/index.tsx (hunk 1 above)
Comment (arielj) [reply src/frontend/components/UI/SearchBar/index.tsx:36]: done!

**Comment by arielj** [comment]
hey @flavioislima I can't merge this, I think it's because I skipped CI in the last commit (because it was just removing a comment), maybe you can merge since you are the owner

**Comment by flavioislima** [comment]
> hey @flavioislima I can't merge this, I think it's because I skipped CI in the last commit (because it was just removing a comment), maybe you can merge since you are the owner

Done

**Comment by Nocccer** [comment]
> hey @flavioislima I can't merge this, I think it's because I skipped CI in the last commit (because it was just removing a comment), maybe you can merge since you are the owner

FYI: You can retrigger Checks INSIDE the pr unser Checks or you use the vscode GitHub Action extension

**Pull request merged**